tests/unit/models/test_projects.py

```diff
@@ -197,12 +197,15 @@ def test_adoptable_field_not_required(self, field, project_yaml_data):
         data = project_yaml_data()
         data.pop(field)
         data["adopt-info"] = "part1"
+        data["parts"] = {"part1": {"plugin": "nil"}}
         project = Project.unmarshal(data)
         assert getattr(project, field) is None
 
     @pytest.mark.parametrize("field", MANDATORY_ADOPTABLE_FIELDS)
     def test_adoptable_field_assignment(self, field, project_yaml_data):
         data = project_yaml_data()
+        data["adopt-info"] = "part1"
+        data["parts"] = {"part1": {"plugin": "nil"}}
```

Comment:
The craft-application project is more strict on the adopt-info value, Snapcraft's implementation didn't really care if the part name was invalid.